CHANGELOG.md

```diff
@@ -2,6 +2,46 @@
 
 <!-- next version -->
 
+## 0.104.0
+
+### 🛑 Breaking changes 🛑
+
+- `opamp`: Adds support for v1beta1 OpenTelemetry Collector API in the OpAMP Bridge (#2985)
+  This change adds support for the OpAMP Bridge to manage and apply OpenTelemetry Collectors using the v1beta1 API in 
+  the OpAMP Bridge. This change removes support for applying OpenTelemetry Collectors using the v1alpha1 API version.
+  The v1beta1 API is the latest version of the OpenTelemetry Collector API and is the recommended version for new 
+  deployments.
+
+### 💡 Enhancements 💡
+
+- `operator`: Release leader election lease on exit (#3058)
+- `collector, target allocator, opamp`: Enabling PodDnsConfig for OpenTelemetry Collector, TargetAllocator and OpAMPBridge. (#2658)
+- `collector`: Make the `spec.mode` field of the `OpenTelemetryCollector` Custom Resource (CR) immutable (#3055)
+- `collector`: Improves the performance of port and configuration parsing in the operator (#2603)
+
+### 🧰 Bug fixes 🧰
+
+- `collector`: Fix deletion issue of `otelcol` CR by making `spec.config.service.pipelines.processors` optional (#3075)
+  This change makes `spec.config.service.pipelines.processors` in `OpenTelemetryCollector` CRD optional, aligning with OTel Collector best practices. It resolves deletion issues by providing flexibility in CRD configuration, addressing conflicts between strict validation and practical uses.
+  Note: Updating the `opentelemetrycollectors.opentelemetry.io` CRD resource is required.
+  
+- `collector`: Allow annotations on service account to prevent infinite reconciliation on OpenShift and creating infinite pull secrets. (#3106)
+  On OpenShift 4.16 the platform automatically adds an annotation `openshift.io/internal-registry-pull-secret-ref: <simplest-collector-dockercfg-jwq66>`
+  to the service account which contains secret name with image pull secret.
+  
+
+### Components
+
+* [OpenTelemetry Collector - v0.104.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.104.0)
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.104.0 

> otlpreceiver: Switch to localhost as the default for all endpoints. (https://github.com/open-telemetry/opentelemetry-collector/issues/8510)

We should at least change docs and think about handling this via upgrade or when we generate config map for the collector.

Reply:
What would you recommend? In case its not specified we switch it explicitly to `0.0.0.0` or set the `$POD_IP`?

Reply:
I'd say `0.0.0.0`, as this was the default up until now.